featuretools/entityset/entity.py

```diff
@@ -75,7 +75,7 @@ def __init__(self, id, df, entityset, variable_types=None,
         if time_index:
             self.set_time_index(time_index, already_sorted=already_sorted)
 
-        entityset.set_secondary_time_index(self, secondary_time_index)
+        self.secondary_time_index = secondary_time_index
```

Comment:
Can this call say as it was? If the validation in `entityset.set_secondary_time_index` is successful it will then set the entity secondary time index to the specified value. Otherwise I think we could be setting it here before the validation check happens.

Reply:
The validation in `set_secondary_time_index` checks if the secondary time index is the same time type across entities. I don't think a Woodwork dataframe will have a reference to the entityset that contains the other dataframes, correct?

Reply:
True, but the handling of the secondary time index is going to change significantly, and all the code here in `entity.py` will be deleted anyway. It's really only an issue in the short term. I think the current code should work fine until that deletion happens though, but not a big deal either way.

Reply:
The only difference for now is that the new code sets the secondary time index on Entity without validation, while the previous code runs validation first and then sets it.

Reply:
I think it'd make sense to keep this validation check outside the entity since it gives an idea of where the validations are likely to happen once the Woodwork DataFrame gets integrated.

Reply:
I view this line as mimicking the user calling `EntitySet.set_secondary_time_index`.  Once Woodwork is added we expect users to call it that way.  In the meantime, `secondary_time_index` is still a valid argument to entity so we need to handle it in some way.

Reply:
Not sure quite sure if I understand the use case, but happy to revert this change

Reply:
✅  7762b431